### PR TITLE
Add LIKE extension LINQ support

### PR DESCRIPTION
**Pull request by onionhammer**
## Description

More low hanging fruit: Provide support for `.Like()` as an extension method, usable within LINQ.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Closing issues

#3106

### CODE REVIEW

**Comment by onionhammer** [comment]
@neildsh @adityasa any thoughts on this one?

**Comment by onionhammer** [comment]
@neildsh @adityasa any comments?

**Comment by adityasa** [comment]
@onionhammer, we are working on adding Regex support which should address some of these scenarios. I would recommend holding off on this PR until that work is complete.

**Comment by onionhammer** [comment]
> @onionhammer, we are working on adding Regex support which should address some of these scenarios. I would recommend holding off on this PR until that work is complete.

Interesting! How does the RU usage compare with LIKE - and why not support both since Cosmos supports both?

**Comment by adityasa** [comment]
@onionhammer - kindly include baseline based e2e coverage (Linq*BaselineTests.cs)

---
In reply to: [1532031145](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3725#issuecomment-1532031145) [](http://example.com/codeflow?ancestors=1532031145,1532031145)

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs [hunk 1]
```diff
@@ -157,6 +157,47 @@ protected override SqlScalarExpression VisitImplicit(MethodCallExpression method
...
+                    new Type[]{typeof(string), typeof(string)},
+                })
+            {
+                this.isNot = isNot;
+            }
+
+            protected override SqlScalarExpression VisitExplicit(MethodCallExpression methodCallExpression, TranslationContext context)
+            {
+                return null;
+            }
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:176]: I believe this is not required.

Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:176]: Is there a specific reason why base class implementation is being overriden?

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/TypeCheckFunctions.cs [hunk 2]
```diff
@@ -44,6 +44,14 @@ static TypeCheckFunctions()
...
                     })
+                },
+                {
+                    "Like",
+                    new StringBuiltinFunctions.StringVisitLike(isNot: false)
+                },
+                {
+                    "NotLike",
+                    new StringBuiltinFunctions.StringVisitLike(isNot: true)
                 }
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/TypeCheckFunctions.cs:55]: These don't really fit as "TypeCheckFunctions"

**Comment by kirankumarkolli** [comment]
Closing inactive PRs older than 3-months, please re-active it needed.

**Comment by onionhammer** [comment]
Lame.

**Pull request closed without merging**